tests_e2e/tests/ext_policy/ext_policy.py

```diff
@@ -0,0 +1,225 @@
+            elif operation == "delete":
+                extension_case.extension.delete(timeout=timeout)
+            fail(f"The agent should have reported an error trying to {operation} {extension_case.extension.__str__()} "
+                 f"because the extension is disallowed by policy.")
+        except Exception as error:
+            assert_that("Extension will not be processed" in str(error)) \
+                .described_as(
+                f"Error message should communicate that extension is disallowed by policy, but actual error "
+                f"was: {error}").is_true()
+            log.info(f"{extension_case.extension.__str__()} {operation} failed as expected")
+
+    def run(self):
+
+        # Prepare extensions to test
+        unique = str(uuid.uuid4())
+        test_file = f"waagent-test.{unique}"
+        custom_script = ExtPolicy.TestCase(
+            VirtualMachineExtensionClient(self._context.vm, VmExtensionIds.CustomScript,
+                                          resource_name="CustomScript"),
+            {'commandToExecute': f"echo '{unique}' > /tmp/{test_file}"}
+        )
+        run_command = ExtPolicy.TestCase(
+            VirtualMachineExtensionClient(self._context.vm, VmExtensionIds.RunCommandHandler,
+                                          resource_name="RunCommandHandler"),
+            {'source': {'script': f"echo '{unique}' > /tmp/{test_file}"}}
+        )
+        unique2 = str(uuid.uuid4())
+        test_file2 = f"waagent-test.{unique2}"
+        run_command_2 = ExtPolicy.TestCase(
+            VirtualMachineExtensionClient(self._context.vm, VmExtensionIds.RunCommandHandler,
+                                          resource_name="RunCommandHandler2"),
+            {'source': {'script': f"echo '{unique2}' > /tmp/{test_file2}"}}
+        )
+        azure_monitor = ExtPolicy.TestCase(
+            VirtualMachineExtensionClient(self._context.vm, VmExtensionIds.AzureMonitorLinuxAgent,
+                                          resource_name="AzureMonitorLinuxAgent"),
+            None
+        )
+
+        # Enable policy via conf
+        log.info("Enabling policy via conf file on the test VM [%s]", self._context.vm.name)
+        self._ssh_client.run_command("update-waagent-conf Debug.EnableExtensionPolicy=y", use_sudo=True)
+
+        # Only allowlisted extensions should be processed.
+        # We only allowlist CustomScript: CustomScript should be enabled, RunCommand and AzureMonitor should fail.
+        policy = \
+            {
+                "policyVersion": "0.1.0",
+                "extensionPolicies": {
+                    "allowListedExtensionsOnly": True,
+                    "signatureRequired": False,
+                    "extensions": {
+                        "Microsoft.Azure.Extensions.CustomScript": {}
+                    }
+                }
+            }
+        self._create_policy_file(policy)
+        self._operation_should_succeed("enable", custom_script)
```

Comment:
Could we also add the test case where single-config ext should fail due to policy

Reply:
Is the test case on line 189 sufficient, or did you mean something else?